docs/getting_started/go/hello_world_in_go/index.md

```diff
@@ -83,419 +85,778 @@ go get go.temporal.io/sdk
 
+```output
+go: downloading go.temporal.io/sdk v1.31.0
+go: added go.temporal.io/sdk v1.31.0
 ```
-go: added go.temporal.io/sdk v1.17.0
+
+With the project created, you'll create the application's core logic.
+
+## Write functions to call external services
+
+Your application will make two HTTP requests. The first request will return your current public IP, while the second request will use that IP to provide city, state, and country information.
+
+You'll use Temporal Activities to make these requests. You use Activities in your Temporal Applications to execute [non-deterministic](https://docs.temporal.io/workflows#deterministic-constraints) code or perform operations that may fail, such as API requests or database calls.
+
+If an Activity fails, Temporal can automatically retry it until it succeeds or reaches a specified retry limit. This ensures that transient issues, like network glitches or temporary service outages, don't result in data loss or incomplete processes.
+
+Create the file `activities.go` in your project root:
+
+```command
+touch activities.go
 ```
 
-With your project workspace configured, you're ready to create your first Temporal Activity and Workflow. You'll start with the Workflow.
+Open the file `activities.go` in your editor and add the following code that imports dependencies and defines a struct to hold the Activities:
 
-## Create a Workflow
 
-Workflows are where you configure and organize the execution of Activities. You write a Workflow using one of the programming languages supported by a Temporal SDK. This code is known as a *Workflow Definition*. 
+<!--SNIPSTART go-ipgeo-activity-setup-->
+[activities.go](https://github.com/temporalio/temporal-tutorial-ipgeo-go/blob/main/activities.go)
+```go
+package iplocate
 
-In the Temporal Go SDK, a Workflow Definition is an [exported function](https://go.dev/tour/basics/3) with two additional requirements: it must accept `workflow.Context` as the first input parameter, and it must return `error`. Your Workflow function can optionally return another value, which you'll use to return the result of the Workflow Execution. You can learn more in the [Workflow parameters](https://docs.temporal.io/dev-guide/go/foundations#workflow-parameters) section of the Temporal documentation.
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type HTTPGetter interface {
+	Get(url string) (*http.Response, error)
+}
+
+type IPActivities struct {
+	HTTPClient HTTPGetter
+}
+
+```
+<!--SNIPEND-->
+
+With the Go SDK, you can define Activities as regular Go functions, or you can create them as members of a struct. If you use a struct, you can then pass in shared objects like database connections or clients that all Activities can share. In this tutorial you'll pass an HTTP client into the struct. This will let you stub out the HTTP client implementation when you write tests for your Activities later.
```

Comment:
I would maybe rewrite this slightly so the choice is very explicit, eg:

```suggestion
With the Go SDK, you can define Activities as regular Go functions. You can also create them as members of a struct, which is necessary to pass shared objects like database connections. In this tutorial you'll pass an HTTP client into the struct. This will let you stub out the HTTP client implementation when you write tests for your Activities later.
```